docs/usage.rst

```diff
@@ -302,6 +302,116 @@ following:
 - ``%q``: The parent task id and description.
 - ``%k``: The current task id and description.
 
+.. _debugging:
+
+Debugging
+=========
+
+Writing task based applications can be tricky, and debugging them even more so. This section
+describes a few options and tools that can be helpful when debugging applications using pika.
+
+.. _segfaults_and_stack_overflows:
+
+Segmentation faults and stack overflows
+---------------------------------------
+
+Due to the small default stack sizes of user level threads a common problem is stack overflows. When
+using the ``mmap``-based stack allocation (default on platforms that support it) pika provides a
+configuration option to enable guard pages at the end of a stack. When enabled, a protected page
+will be allocated such that if one attempts to read or write within a page of the end of the stack,
+a segmentation fault will be triggered. The option can be enabled by exporting
+``PIKA_USE_GUARD_PAGES=1``. The option is disabled by default.
+
+.. info::
+
+   Many MPI implementations also install signal handlers or have options for enabling them.
+   ``libSegFault.so`` (part of ``glibc-tools``, more info in `this blog post
+   <https://www.marcusfolkesson.se/blog/libsegfault/>`_) also provides a way to install a signal
+   handler. These can be useful alternatives to the signal handlers provided by pika. Depending on
+   the type of issue you are debugging, different signal handlers can be more or less helpful as
+   they print slightly different information.
+
+   pika does not install any signal handlers by default. They have to be enabled explicitly using
+   the options described below.
```

Comment:
This note feels misplaced: should it go after the next session? Mainly asking because of the "also install", but you only says below that pika can install signal handlers.

Reply:
Yeah, I definitely put it too early. I've moved it down a bit in https://github.com/pika-org/pika/pull/1412/commits/33e656a8e0f6196239f37ea386a06cb553e4aaab.